src/Glpi/Controller/GenericFormController.php

```diff
@@ -0,0 +1,193 @@
+use Html;
+use Glpi\Event;
+use Glpi\Exception\Http\AccessDeniedHttpException;
+use Glpi\Exception\Http\BadRequestHttpException;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+use Toolbox;
+
+class GenericFormController extends AbstractController
+{
+    public const ACTIONS_AND_CHECKS = [
+        'get' => ['permission' => READ],
+        'add' => ['permission' => CREATE],
+        'delete' => ['permission' => DELETE],
+        'restore' => ['permission' => DELETE],
+        'purge' => ['permission' => PURGE],
+        'update' => ['permission' => UPDATE],
+        'unglobalize' => ['permission' => UPDATE],
+    ];
+
+    #[Route("/{class}/Form", name: "glpi_itemtype_form", priority: -1)]
+    public function __invoke(Request $request): Response
+    {
+        $class = $request->attributes->getString('class');
+
+        $this->checkIsValidClass($class);
+
+        /** @var class-string<CommonDBTM> $class */
+
+        if (!$class::canView()) {
+            throw new AccessDeniedHttpException();
+        }
+
+        $form_action = $this->getCurrentAllowedAction($request, $class);
+
+        if (!$form_action) {
+            throw new AccessDeniedHttpException();
+        }
+
+        if ($response = $this->handleFormAction($request, $form_action, $class)) {
+            return $response;
+        }
+
+        throw new NotFoundHttpException();
+    }
+
+    private function checkIsValidClass(string $class): void
+    {
+        if (!$class) {
+            throw new BadRequestHttpException('The "class" attribute is mandatory for dropdown routes.');
+        }
+
+        if (!\class_exists($class)) {
+            throw new BadRequestHttpException(\sprintf("Class \"%s\" does not exist.", $class));
+        }
+
+        if (!\is_subclass_of($class, CommonDBTM::class)) {
+            throw new BadRequestHttpException(\sprintf("Class \"%s\" is not a DB object.", $class));
+        }
+    }
+
+    /**
+     * @param class-string<CommonDBTM> $class
+     */
+    private function handleFormAction(Request $request, string $form_action, string $class): ?Response
+    {
+        $id = $request->query->get('id', -1);
+        $object = new $class();
+        $post_data = $request->request->all();
+        $isTemplateForm = ($object->maybeTemplate() && $object->isTemplate()) || $request->query->get('withtemplate');
+
+        if (!$object instanceof \CommonDBTM) {
+            Event::log(
+                '-',
+                \strtolower(\basename($class)),
+                $object::getLogLevel(),
+                $object::getLogServiceName(),
+                sprintf(__('%1$s tried to execute the "%2$s" action on the item %3$s, but this type does not support POST actions.'), $_SESSION["glpiname"], $form_action, $post_data["name"])
+            );
+
+            return null;
+        }
```

Comment:
This will never happen. `checkIsValidClass` will only accept `CommonDBTM` classes.

```suggestion
```